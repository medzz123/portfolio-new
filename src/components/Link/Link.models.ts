import { ComponentProps } from '@stitches/react';
import { FontSizes, FontWeights } from '@theme/font';
import { LinkProps as NextLinkProps } from 'next/link';

import { StyledLink } from './Link.styles';

export interface LinkProps extends NextLinkProps {
  weight?: FontWeights;
  size?: FontSizes;
  variant?: 'line' | 'box';
  htmlLinkProps?: ComponentProps<typeof StyledLink>;
}
