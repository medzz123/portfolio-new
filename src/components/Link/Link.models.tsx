export interface LinkProps {
  href: string;
  variant?: 'primary' | 'secondary';
}
