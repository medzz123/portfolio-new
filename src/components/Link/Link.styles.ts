import { styled } from '@theme/config';

export const StyledLink = styled('a', {
  color: '$primary',
  fontSize: '$info',
  cursor: 'pointer',
  '&:hover': {
    textDecoration: 'underline',
  },

  variants: {
    type: {
      primary: {
        color: '$primary',
      },
      secondary: {
        color: '$hiContrast',
      },
    },
  },
});
