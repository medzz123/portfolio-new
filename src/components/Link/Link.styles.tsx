import { styled } from '@theme/config';

export const StyledLink = styled('a', {
  cursor: 'pointer',
  textDecoration: 'none',
  position: 'relative',
  color: '$hiContrast',
  overflow: 'hidden',
  variants: {
    type: {
      line: {
        fontWeight: '$700',
        '&::before': {
          position: 'absolute',
          content: '',
          width: '$full',
          height: '$3',
          backgroundColor: '$accent',
          left: 0,
          bottom: 0,
        },
        '&::after': {
          position: 'absolute',
          content: '',
          width: '$12',
          height: '$3',
          backgroundColor: '$loContrast',
          left: '-$12',
          bottom: 0,
        },
      },
      box: {},
    },
  },
});
