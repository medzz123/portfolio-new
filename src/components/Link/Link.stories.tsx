import { Meta, Story } from '@storybook/react/types-6-0';
import React from 'react';

import Link from './Link';

export default {
  title: 'Components/Link',
  component: Link,
} as Meta;

const Template: Story = (storyArguments) => (
  <Link href="#" {...storyArguments}>
    Hello
  </Link>
);

export const Default = Template.bind({});
