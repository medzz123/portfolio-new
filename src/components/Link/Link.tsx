import NextLink from 'next/link';
import React, { FunctionComponent } from 'react';

import { LinkProps } from './Link.models';
import { StyledLink } from './Link.styles';

const Link: FunctionComponent<LinkProps> = (props) => {
  const { children, href, variant = 'primary' } = props;

  return (
    <NextLink href={href}>
      <StyledLink data-testid="link" type={variant}>
        {children}
      </StyledLink>
    </NextLink>
  );
};

export default Link;
