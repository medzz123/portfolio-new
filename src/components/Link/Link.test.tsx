import { render } from '@test/library';
import React from 'react';

import Link from './Link';

test('Link Test', () => {
  const { getByTestId, container } = render(<Link href="#">Hello</Link>);

  const linkNode = getByTestId('link');

  expect(linkNode.textContent).toBe('Hello');

  expect(container).toMatchSnapshot();
});
