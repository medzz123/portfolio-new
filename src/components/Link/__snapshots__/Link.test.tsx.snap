// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Test 1`] = `
<div>
  <a
    class="sx28ds4 sx28ds44nwvh--type-primary"
    data-testid="link"
  >
    Hello
  </a>
</div>
`;
